packages/sdk/src/evm/core/sdk.ts

```diff
@@ -185,7 +185,7 @@ export class ThirdwebSDK extends RPCConnectionHandler {
     storage?: ThirdwebStorage,
   ): ThirdwebSDK {
     let signerWithProvider = signer;
-    if (network && !signer.provider) {
+    if (network) {
```

Comment:
Careful with this, some signers throw if you try to override the provider. That's the case for most injected browser wallets which is why this check was here. We only override provider for signers like private key / kms / backend wallets etc which don't come with providers so we need to put one in. 

Network should probably be optional for this static method tbh

Reply:
Gotcha - `network` is already optional - but the interface is a bit misleading. We could also catch the error - that might be the best approach

Reply:
This led to a bug in web3api where network gets passed through but doesn't get respected

Reply:
Try catch should be ok